x-pack/plugins/apm/public/components/shared/charts/instances_latency_distribution_chart/custom_tooltip.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TooltipInfo } from '@elastic/charts';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { getServiceNodeName } from '../../../../../common/service_nodes';
+import {
+  asTransactionRate,
+  TimeFormatter,
+} from '../../../../../common/utils/formatters';
+import { useTheme } from '../../../../hooks/use_theme';
+import { PrimaryStatsServiceInstanceItem } from '../../../app/service_overview/service_overview_instances_chart_and_table';
+
+const latencyLabel = i18n.translate(
+  'xpack.apm.instancesLatencyDistributionChartTooltipLatencyLabel',
+  {
+    defaultMessage: 'Latency',
+  }
+);
+
+const throughputLabel = i18n.translate(
+  'xpack.apm.instancesLatencyDistributionChartTooltipThroughputLabel',
+  {
+    defaultMessage: 'Throughput',
+  }
+);
+
+/**
+ * Tooltip for a single instance
+ */
+function SingleInstanceCustomTooltip({
+  latencyFormatter,
+  values,
+}: TooltipInfo & { latencyFormatter: TimeFormatter }) {
+  const value = values[0];
+  const { color } = value;
+  const datum = (value.datum as unknown) as PrimaryStatsServiceInstanceItem;
+  const { latency, serviceNodeName, throughput } = datum;
+
+  return (
+    <>
+      <div className="echTooltip__header">
```

Comment:
Is this the recommended way of creating custom tooltips? Maybe it's just me but `classname` seems like such an ancient concept :D

Reply:
Perhaps we should make the react components ourselves first.

Reply:
This is the way to do it if you want to make it look like existing tooltips without having to implement all the styles yourself. I put a comment on the Elastic Charts tooltip improvement issue mentioning that this is not the best: https://github.com/elastic/elastic-charts/issues/615#issuecomment-810471651